src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2062,8 +2062,8 @@ codet java_bytecode_convert_methodt::convert_instructions(
       // (value1 == NaN || value2 == NaN) ?
       //   nan_value : value1 == value2 ? 0  : value1 < value2 -1 ? 1 : 0;
 
-      exprt nan_op0=ieee_float_equal_exprt(nan_expr, op[0]);
-      exprt nan_op1=ieee_float_equal_exprt(nan_expr, op[1]);
+      isnan_exprt nan_op0(op[0]);
```

Comment:
please also remove `      const constant_exprt nan_expr(nan.to_expr());` as it's not used any more